doc/source/ray-core/walkthrough.rst

```diff
@@ -2,125 +2,144 @@
+and ``get_progress()`` method to get the overall progress.
+
+Defining the Sampling Task
+--------------------------
+Next, we will define a Ray task that does the sampling up to the given times and returns the number of samples that are inside the circle.
+Ray tasks are stateless functions that will be executed asynchronously and in parallel.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __defining_task_start__
+    :end-before: __defining_task_end__
+
+We define a Ray task by decorating a normal Python function with :ref:`ray.remote() <ray-remote-ref>`.
+The sampling task takes a progress actor handle as an input and reports progress to it.
+This shows an example of calling actors from tasks.
+
+Creating a Progress Actor
+-------------------------
+Once the actor is defined, we can create an instance of it.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __creating_actor_start__
+    :end-before: __creating_actor_end__
+
+We create an instance of the progress actor by calling ``remote()`` method with arguments to the constructor.
+This will create and run the actor on a remote worker process.
+The return value of ``Actor.remote(...)`` is an actor handle that can be used to call the actor later on.
+
+Executing Sampling Tasks
+------------------------
+Once the task is defined, we can execute it asynchronously.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __executing_task_start__
+    :end-before: __executing_task_end__
+
+We execute the sampling task by calling ``remote()`` method with arguments to the function.
+This will immediately return an ``ObjectRef`` as a future
+and then execute the function asynchronously on a remote worker process.
+
+Calling the Progress Actor
+--------------------------
+While sampling tasks are running, we can periodically query the progress by calling the actor ``get_progress()`` method.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __calling_actor_start__
+    :end-before: __calling_actor_end__
+
+We can call actor method via ``actor.method.remote()``.
+This will immediately return an ``ObjectRef`` as a future
+and then execute the method asynchronously on the remote actor process.
+After that, we use the blocking :ref:`ray.get() <ray-get-ref>` to get the actual return value that ``ObjectRef`` represents.
+
+Calculating π
+-------------
+Finally, we get number of samples inside the circle returned from sampling tasks and calculate π.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __calculating_pi_start__
+    :end-before: __calculating_pi_end__
+
+As we can see, besides a single ``ObjectRef``, :ref:`ray.get() <ray-get-ref>` can also take a list of ``ObjectRef`` and return a list of results.
+
+Shutting Down Ray
```

Comment:
Let's remove this section. Shutting down Ray is not something users ever do (and definitely not in getting started).